### ISSUE TITLE
domus.Link Android fails with error:  "Error communicating with Domus:  Problem parsing API response (\n) Update Settings or Exit"

### ISSUE THREAD

**Issue by latenitetech** [bwssytems/domus.Link.Android#1]
I downloaded and installed the currently available domus.Link Android app from the Google Play Store.  In settings, I entered my domus.Link host as "192.168.1.1:8010", and entered the current (still default) password of "1234."  Upon exiting settings, I get the error:
_"Error communicating with Domus:  Problem parsing API response
Update Settings or Exit"_

My Android devices (same error on both):
Samsung Galaxy S4, running Android version 5.0.1
Asus ZenPad (P01M) tablet, running Android version 5.0

Server side: 
heyu and web server running on Asus RT-AC68U router w/AsusWRT-Merlin firmware.  Web server is lighttpd.  This config works fine serving up domus.Link to PC-based web browers (Firefox and Chrome), and the "mobileWebKit" interface to Chrome on my mobile devices is working as well.

I don't see any errors in the lighttpd error log.  I'd be glad to provide any additional diagnostics info if you point me to how to collect it.

Thx!

**Comment by bwssytems** [comment]
Ahh, Just a simple question, are you on your local network when you do this? As in your S4 is connected to your home wifi.

**Comment by latenitetech** [comment]
Yes, I've not enabled WAN access to my domus.Link web server (lighttpd), so
all access is via my local LAN.  My S4 (running Chrome) can get to the web
page at 'http://192.168.1.1:8010/' just fine, and the domus.Link mobile
view works fine that way.

**Comment by bwssytems** [comment]
Do you have a lot of Aliases configured already? Maybe if you pair it down to have one alias to test as the parsing occurs on getting the first list of aliases from the server.

**Comment by latenitetech** [comment]
I did a little more looking and believe the problem may be somewhere in setupverify.php.  I noticed whenever I invoked domus.Link from a web browser, I was always presented with the 'Setup Verification' screen, with a table of my 11 heyu-defined devices showing like this:
![image](https://cloud.githubusercontent.com/assets/24569293/22353135/a6aa1af2-e3e4-11e6-9918-2107f334e778.png)
even though they're all already defined in my aliasmap file (I added the .txt suffix here so I could paste it into this entry):
[aliasmap.txt](https://github.com/bwssytems/domus.Link.Android/files/733767/aliasmap.txt)
I've just been hitting 'Continue' and everything seems to work fine from that point on.  Looking at the code, in index.php, the check at line 75 is always failing, so setupverify is called, and that leg of code always ends up presenting the table above as if it's a first-time run.  I don't see that issue when I use a web browser on my mobile device.

My x10.conf file is pretty simple, but I'm including it here in case you see something in it that's tripping up domus.Link (again, added the .txt suffix so it would paste here):
[x10.conf.txt](https://github.com/bwssytems/domus.Link.Android/files/733804/x10.conf.txt)

Thoughts?

**Comment by bwssytems** [comment]
have you tried hitting convert, it is looking like you have an old format of devices

**Comment by latenitetech** [comment]
I've tried to "convert" several times.  Not sure what it's supposed to do.
What I observe happening is:
1)  re-writes the heyu x10,conf file, but it appears nothing has changed
2)  it throws away the previous aliasmap file and populates a new one that
matches the "verification" table
I can then use the setup menu to update the alias file and save it, but on
the next browsing session the cycle starts over again, as if it's again a
"first time startup" even though the modified aliasmap file is there.

I don't know what you mean by "old format of devices."  As a relative
noobie to heyu, I populated my initial x10.conf in the past month based
simply on the comments in the sample file, and man pages provided with the
heyu distribution.  If there is a more current format, I don't have any
idea what that is.  Or does domus.Link impose additional constraints over
how the x10.conf file is populated?  For instance, how does domus.Link
derive "Location" from the heyu x10.conf file?  I've never seen a reference
to a device location in the heyu documentation, although it makes complete
sense in the context of domus.Link.

I do appreciate your help here, but it appears something strange is
happening with my config.
